### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by vanyachyzh**
[DEMO LINK](https://vanyachyzh.github.io/react_pagination/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [total] = useState(items);
```
Comment (Viktor-Kost) [inline src/App.tsx:11]: it looks like a state for items, not for total

File: src/App.tsx [hunk 2]
```diff
@@ -1,18 +1,42 @@
...
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(0);
+
+  const getFirstRow = (amount: number, position: number) => (
+    position * amount + 1
+  );
```
Comment (Viktor-Kost) [inline src/App.tsx:17]: move this function outside the component

File: src/App.tsx [hunk 3]
```diff
@@ -1,18 +1,42 @@
...
+
+  const getLastRow = (
+    amount: number,
+    currentPag: number,
+    totalAmount: number,
+  ) => {
+    const lastRow = currentPag * amount + amount;
+
+    return lastRow > totalAmount ? totalAmount : lastRow;
+  };
```
Comment (Viktor-Kost) [inline src/App.tsx:27]: move as well

File: src/App.tsx [hunk 4]
```diff
@@ -1,18 +1,42 @@
...
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [total] = useState(items);
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(0);
```
Comment (Viktor-Kost) [inline src/App.tsx:13]: how page can be with a 0 value? it should start from 1

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
Pretty nice. Need some improvements

File: src/App.tsx [hunk 5]
```diff
@@ -1,18 +1,28 @@
...
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
+const itemsList = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [items] = useState(itemsList);
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
```
Comment (sTorba24) [inline src/App.tsx:13]: you can store start values in variables to not use 'magic values'

File: src/components/Pagination/Pagination.tsx [hunk 6]
```diff
@@ -1 +1,111 @@
...
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const book = splitPages(items, perPage);
+  const numOfPages = Object.keys(book);
+  const numLastPage = numOfPages.length;
+
+  const getNextPage = () => {
+    if (currentPage === items.length - 1) {
```
Comment (sTorba24) [inline src/components/Pagination/Pagination.tsx:23]: seems you are comparing page number and length of items array(42)?
You should check if page is not last, so currentPage === lastPage?

File: src/components/Pagination/Pagination.tsx [hunk 7]
```diff
@@ -1 +1,111 @@
...
+      return;
+    }
+
+    onPageChange(prevPage => prevPage + 1);
+  };
+
+  const getPrevPage = () => {
+    if (currentPage === 0) {
```
Comment (sTorba24) [inline src/components/Pagination/Pagination.tsx:31]:
```suggestion
    if (currentPage === 1) {
```

Comment (sTorba24) [reply src/components/Pagination/Pagination.tsx:31]: Current page can be 0?

File: src/components/Pagination/Pagination.tsx [hunk 8]
```diff
@@ -1 +1,111 @@
...
+
+  return (
+    <>
+      <ul className="pagination">
+        <li
+          key="prev"
+          className={classNames(
+            'page-item',
+            {
+              disabled: currentPage === 1,
```
Comment (sTorba24) [inline src/components/Pagination/Pagination.tsx:46]: you can use variable like `isLastPage`

File: src/components/Pagination/Pagination.tsx [hunk 9]
```diff
@@ -1 +1,111 @@
...
+            </a>
+          </li>
+        ))}
+
+        <li
+          key="next"
+          className={classNames(
+            'page-item',
+            {
+              disabled: currentPage === numLastPage,
```
Comment (sTorba24) [inline src/components/Pagination/Pagination.tsx:89]: `isLastPage`

**Review by sTorba24 — CHANGES_REQUESTED** [review]
Good work! Some suggestions to improve it :)

**Review by IvanFesenko — APPROVED** [review]
Well done

**Pull request closed without merging**